File: python_version/dblank.py
class StringUtils:
    @staticmethod
    def dblank(s):
        """
        Returns the number of characters up to and including the last non-blank character.
        Equivalent to Fortran's dblank.
        """
        ns = len(s)
        while ns > 1 and s[ns-1] == ' ':
            ns -= 1
        return ns

# Example usage:
# s = "hello   "
# ns = StringUtils.dblank(s)
# print(ns)  # Output: 5

File: model/dblank.f
      subroutine dblank(s, ns)
c
c
c  @(#) dblank.f  McKie  Jun-1988
c  This routine finds index of last nonblank char in a string.
c
c  Argument list input:
c    s = string to be examined
c
c  Argument list output:
c    ns = # chars in s, up to & including last non-blank
c
c
c  Declare subprogram arg(s)
c
      character*(*) s
      integer ns
c
c
c  Find last non-blank char in string (beyond 1st char), return to caller
c
      ns = len(s)
 2100 if( ( s(ns:ns) .eq. ' ' ) .and. ( ns .gt. 1 ) )then
       ns = ns - 1
       goto 2100
      endif
      return
      end
